src/mapdata.cpp

```diff
@@ -300,6 +300,32 @@ bool map_deconstruct_info::load( const JsonObject &jsobj, const std::string &mem
     return true;
 }
 
+ter_shoot_info::ter_shoot_info() : chance_to_hit( 0.0f ),
+    reduce_dmg_min( 0 ), reduce_dmg_max( 0 ),
+    reduce_dmg_min_laser( 0 ), reduce_dmg_max_laser( 0 ),
+    destroy_dmg_min( 0 ), destroy_dmg_max( 0 ),
+    laser_can_destroy( false ) {}
+
+bool ter_shoot_info::load( const JsonObject &jsobj, const std::string &member )
+{
+    JsonObject j = jsobj.get_object( member );
+
+    assign( j, "chance_to_hit", chance_to_hit );
+
+    assign( j, "reduce_dmg_min", reduce_dmg_min );
+    assign( j, "reduce_dmg_max", reduce_dmg_max );
+
+    assign( j, "reduce_dmg_min_laser", reduce_dmg_min_laser );
+    assign( j, "reduce_dmg_max_laser", reduce_dmg_max_laser );
+
+    assign( j, "destroy_dmg_min", destroy_dmg_min );
+    assign( j, "destroy_dmg_max", destroy_dmg_max );
+
+    assign( j, "laser_can_destroy", laser_can_destroy );
```

Comment:
Please use optional or mandatory here instead.

Reply:
Terrain doesn't use copy-from (or doesn't with proportional/relative)?

Reply:
More that I'd like to remove those at some point in the near future, and so want to make sure less uses them before that.

Reply:
I assume for "was_loaded" I use the was_loaded member from ter_t?

Reply:
Yes.